### ISSUE TITLE
EventHandlers are being called from non-main-thread

### ISSUE THREAD

**Issue by leoHN** [googleads/googleads-mobile-unity#821]
I create this post to tell someone know all of EventHandlers are being called from other threads (OnAdLoaded, OnAdClosed...). It may not be a bug. But it causes some bugs, such as "StopAllCoroutines can only be called from the main thread." I use a trick to fix it temporarily. Check it if you need. [https://stackoverflow.com/questions/41330771/use-unity-api-from-another-thread-or-call-a-function-in-the-main-thread](url).

**Comment by juyujuyu** [comment]
Yes, it's easy to causes some bugs, I do not know why googleads do not fix it.
To fix it, I use Action and put all callback into it, and call them form MonoBehaviour.Update.

**Comment by Antik22** [comment]
F*ck, I lost around 3-4 hours with this issue until I finally understood that it's not main thread and that's reason why it crashes.
